### PR TITLE
Improve the oauth2 implementation

### PR DESCRIPTION
**Pull request by zymap**
*Motivation*

When using pulsarctl as a library, the oauth2 auth provider should initialize the authenticate flow and save the token to the store. When using pulsarctl as a command-line tool, users should login or active at first to get access token and save it to the store, then other commands can get the token from the store.

*Modification*

- Add login command
- Add activate command
- Update the oauth2Porvider code

### CODE REVIEW

**Comment by zymap** [comment]
Replace the oauth2 package as soon as the code merge to pulsar-client-go.

**Comment by zymap** [comment]
@EronWright  Sure. please go ahead. Thanks.

File: pkg/oauth2/active.go [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+
+func activateCmd(vc *cmdutils.VerbCmd) {
+	desc := cmdutils.LongDescription{}
+	desc.CommandUsedFor = "This command is used for activating a service account by supplying its credentials."
+	desc.CommandPermission = "This command doesn't need pulsar permissions."
+
+	var examples []cmdutils.Example
+	activate := cmdutils.Example{
+		Desc:    "Activate a service account by supplying its credentials.",
+		Command: "pulsarctl oauth2 active --issuer-endpoint (issuer) --audience (audience) --key-file (key file path)",
```
Comment (EronWright) [inline pkg/oauth2/active.go:37]:
```suggestion
		Command: "pulsarctl oauth2 activate --issuer-endpoint (issuer) --audience (audience) --key-file (key file path)",
```

File: pkg/oauth2/active.go [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+
+	vc.SetRunFunc(func() error {
+		return doActive(vc, issuerEndpoint, audience, keyFile)
+	})
+
+	vc.FlagSetGroup.InFlagSet("Oauth2 login", func(set *pflag.FlagSet) {
+		set.StringVarP(&issuerEndpoint, "issuer-endpoint", "i", "",
+			"The OAuth 2.0 issuer endpoint")
+		set.StringVarP(&audience, "audience", "a", "",
+			"The audience identifier for the API server")
```
Comment (EronWright) [inline pkg/oauth2/active.go:59]:
```suggestion
			"The audience identifier for the Pulsar instance")
```

File: pkg/oauth2/active.go [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+		set.StringVarP(&keyFile, "key-file", "k", "",
+			"Path to the private key file")
+	})
+}
+
+func doActive(vc *cmdutils.VerbCmd, issuerEndpoint, audience, keyFile string) error {
```
Comment (EronWright) [inline pkg/oauth2/active.go:65]: Please rename to `doActivate`

File: pkg/oauth2/login.go [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+
+	vc.SetRunFunc(func() error {
+		return doLogin(vc, issuerEndpoint, clientID, audience, false)
+	})
+
+	vc.FlagSetGroup.InFlagSet("Oauth2 login", func(set *pflag.FlagSet) {
+		set.StringVarP(&issuerEndpoint, "issuer-endpoint", "i", "",
+			"The OAuth 2.0 issuer endpoint")
+		set.StringVarP(&audience, "audience", "a", "",
+			"The audience identifier for the API server")
```
Comment (EronWright) [inline pkg/oauth2/login.go:61]:
```suggestion
			"The audience identifier for the Pulsar instance")
```

**Pull request merged**